mission_ctl.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+func (r *regolancer) addFailedChan(fromStr string, toStr string, amount int64) {
+	r.mcCache[fromStr+toStr] = amount
+}
+
+func (r *regolancer) validateRoute(route *lnrpc.Route) error {
+	lastHopPK := ""
+	for _, h := range route.Hops {
+		hopPK := h.PubKey
+		if lastHopPK == "" {
+			lastHopPK = hopPK
+			continue
+		}
+		if fp, ok := r.mcCache[lastHopPK+hopPK]; ok && 1e6-h.AmtToForwardMsat*1e6/fp < params.FailTolerance {
+			from, err := hex.DecodeString(lastHopPK)
+			if err != nil {
+				return err
+			}
+			to, err := hex.DecodeString(hopPK)
+			if err != nil {
+				return err
+			}
+			r.failedPairs = append(r.failedPairs, &lnrpc.NodePair{From: from, To: to})
+			return fmt.Errorf("chan %d failed before with %d msat, payment attempt with %d msat cancelled", h.ChanId, fp, h.AmtToForwardMsat)
```

Comment:
maybe add in the output that all further payments of the directed channel will be failed too? So that nobody will wonder in the future

Reply:
Added!